pkg/debug/checkups/osquery.go

```diff
@@ -0,0 +1,147 @@
+package checkups
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+)
+
+type osqueryCheckup struct {
+	status         Status
+	executionTimes map[string]string // maps command to how long it took to run, in ms
+	summary        string
+}
+
+func (o *osqueryCheckup) Name() string {
+	return "Osquery"
+}
+
+func (o *osqueryCheckup) Run(ctx context.Context, extraWriter io.Writer) error {
+	// Determine passing status by running osqueryd --version
+	o.executionTimes = make(map[string]string)
+	if osqueryVersion, err := o.version(ctx); err != nil {
+		o.status = Failing
+		o.summary = err.Error()
+		return fmt.Errorf("running osqueryd version: %w", err)
+	} else {
+		o.status = Passing
+		o.summary = osqueryVersion
+	}
+
+	// Run launcher interactive to capture timing details
+	if err := o.interactive(ctx); err != nil {
+		return fmt.Errorf("running launcher interactive: %w", err)
+	}
+
+	// If we're running doctor and not flare, that's all we need
+	if extraWriter == io.Discard {
+		return nil
+	}
+
+	zipWriter := zip.NewWriter(extraWriter)
+	defer zipWriter.Close()
```

Comment:
I don't know if you need this at all. Though you add it to capture stdout on earlier two commands